topic/tests/index.md

```diff
@@ -508,4 +508,22 @@ Sometimes a directive may call an internal function that performs specific actio
 With the above example, we could combine the data sets with this example to provide even shorter code, but it may be undesirable to do so.  For instance, with this code structure one can assert additional effects happened or mix in several `beforeEach()` calls.  Remember that the goal is maintainable, clean and self-describing code.
 
 
+Functional Testing
+------------------
+
+Functional testing is the art of testing how your website or api works from the eye of a user, or at least with everything put together. It is very similar to unit testing in the way of how tests are laid out, but there are some things to watch out for.
+
+First, we want to load the page or refresh each time a test is started. This is to make sure we carry over any data from one test to another. For instance, if data was set to be used in the `session` we want to make sure it gets cleared out before loading the next test. This can usually be done in the `afterEach()` call using `window.sessionStorage.clear();`.
+
+Try to think what a user would do. Try running through the scenario you are setting up and pay attention to what you do. Forms can be a little touchy, as some people will click the button to submit the form, while others will tab to the submit button and hit enter, or just hit enter when in the last field. Try to write tests using different techniques to achieve the same goal, like submitting a form to make sure different ways work.
+
+Make sure browsers you are using are set to 100% zoom and your Operating System (OS) is not zooming past 100%. Internet Explorer is fickle in both of these. When writing your functional tests make sure both are set to 100%. If you need to change the OS zoom level, make sure to check the browsers are both set to 100% before trying to test. This can interfere with the ability of using a click method.
```

Comment:
This sounds like wonderful advice, but only applies to UIs.  Maybe this paragraph and the next should move to a subheading?

    ### Functional Testing of UIs Rendered In Browsers